lib/src/client_offline_mixin.dart

```diff
@@ -0,0 +1,662 @@
+            final cacheModel = queuedWrite['cacheModel'] as String;
+            final cacheKey = queuedWrite['cacheKey'] as String;
+            final modelStore = getModelStore(cacheModel);
+            if (previous != null) {
+              await upsertCache(txn, modelStore, previous, key: cacheKey);
+            }
+          });
+        }
+      } catch (e) {
+        if (onError != null) {
+          onError(e);
+        }
+      }
+    }
+  }
+
+  bool resultIsOnline(ConnectivityResult result) {
+    return result == ConnectivityResult.mobile ||
+        result == ConnectivityResult.ethernet ||
+        result == ConnectivityResult.wifi;
+  }
+
+  Future<void> checkOnlineStatus() async {
+    try {
+      final url = Uri.parse('https://appwrite.io/version');
+      await http.head(url).timeout(Duration(seconds: 1));
+      isOnline.value = true;
+    } catch (_) {
+      isOnline.value = false;
+    }
+  }
+
+  Future<void> listenForConnectivity() async {
+    void handleConnectivityResult(ConnectivityResult result) {
+      isOnline.value = resultIsOnline(result);
+    }
+
+    final result = await Connectivity().checkConnectivity();
+    handleConnectivityResult(result);
+    if (isOnline.value) {
+      // wifi or mobile is connected, but double check internet connectivity
+      await checkOnlineStatus();
+    }
+    Connectivity().onConnectivityChanged.listen(handleConnectivityResult);
+  }
+
+  Future<Response> handleOfflineRequest({
+    required Uri uri,
+    required HttpMethod method,
+    required Future<Response<dynamic>> Function(HttpMethod,
+            {String path,
+            Map<String, String> headers,
+            Map<String, dynamic> params,
+            ResponseType? responseType,
+            String cacheModel,
+            String cacheKey,
+            String cacheResponseIdKey,
+            String cacheResponseContainerKey})
+        call,
+    String path = '',
+    Map<String, String> headers = const {},
+    Map<String, dynamic> params = const {},
+    ResponseType? responseType,
+    String cacheModel = '',
+    String cacheKey = '',
+    String cacheResponseIdKey = '',
+    String cacheResponseContainerKey = '',
+    Map<String, Object?>? previous,
+  }) async {
+    if (!headers.containsKey('X-Appwrite-Timestamp')) {
+      headers['X-Appwrite-Timestamp'] =
+          DateTime.now().toUtc().toIso8601String();
+    }
+    final pathSegments = uri.pathSegments;
+    String queuedWriteKey = '';
+
+    final store = getModelStore(cacheModel);
+    switch (method) {
+      case HttpMethod.get:
+        if (cacheKey.isNotEmpty) {
+          final recordRef = store.record(cacheKey);
+          final record = await recordRef.get(db);
+          if (record != null) {
+            updateAccessedAt(db, store.name, cacheKey);
+            return Response(data: record);
+          }
+        } else {
+          final finder = Finder(limit: 25);
```

Comment:
Can we avoid hard coding the 25 limit here?

Reply:
Would declaring a constnat in the mixin suffice?

Reply:
Yep should do, not sure of any good way we can get the default limit from the spec 👍